File: dart/grains.dart
import 'dart:math';

BigInt square(final int n) {
  if (n < 1 || n > 64) {
    throw ArgumentError("square must be between 1 and 64");
  }
  return BigInt.from(2).pow(n - 1);
}

BigInt total() {
  return BigInt.from(2).pow(64) - BigInt.from(1);
}

File: zig/grains.zig
const math = @import("std").math;

pub const ChessboardError = error{
    IndexOutOfBounds,
};

pub fn square(index: usize) ChessboardError!u64 {
    if (index == 0 or index > 64) {
        return ChessboardError.IndexOutOfBounds;
    }
    const i: u6 = @intCast(index - 1);
    return @as(u64, 1) << i;
}

pub fn total() u64 {
    return math.maxInt(u64);
}
